src/components/processes-configuration.js

```diff
@@ -105,61 +105,35 @@ const keyGenerator = (() => {
     return () => key++;
 })();
 
-const ProcessTsos = ({
-    initialTsos,
-    processIndex,
-    handleProcessTsosChanged,
-}) => {
+const ProcessTsos = ({ tsosList, processIndex, handleProcessTsosChanged }) => {
     const intl = useIntl();
-    // processTsos copies will be deleted in an upcoming PR
-    const [processTsos, setProcessTsos] = useState(
-        initialTsos.map((e) => {
-            return {
-                id: keyGenerator(),
-                sourcingActor: e,
-            };
-        })
-    );
-
-    useEffect(() => {
-        handleProcessTsosChanged(processIndex, processTsos);
-        // Do not add handleProcessTsosChanged as dependency to avoid infinite loop
-        // To be changed
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [processTsos]);
 
     const handleTsoSourcingActorChanged = (newValue, index) => {
-        const processTsosCopy = [...processTsos];
-        processTsosCopy[index] = {
-            id: processTsosCopy[index].id,
-            sourcingActor: newValue,
-        };
-        setProcessTsos(processTsosCopy);
+        const processTsosCopy = [...tsosList];
+        processTsosCopy[index] = newValue;
+        handleProcessTsosChanged(processIndex, processTsosCopy);
     };
 
     const handleAddProcessTso = () => {
-        const processTsosCopy = [...processTsos];
-        processTsosCopy.push({
-            id: keyGenerator(),
-            sourcingActor: '',
-        });
-        setProcessTsos(processTsosCopy);
+        const processTsosCopy = [...tsosList];
+        processTsosCopy.push('');
+        handleProcessTsosChanged(processIndex, processTsosCopy);
     };
 
     const handleRemoveProcessTso = (index) => {
-        const processTsosCopy = [...processTsos];
+        const processTsosCopy = [...tsosList];
         processTsosCopy.splice(index, 1);
-        setProcessTsos(processTsosCopy);
+        handleProcessTsosChanged(processIndex, processTsosCopy);
     };
 
     return (
         <>
-            {processTsos.map((tso, index) => (
-                <Grid container spacing={2} key={tso.id}>
+            {tsosList.map((tso, index) => (
+                <Grid container spacing={2} key={tso}>
```

Comment:
when you click twice on the add button in row (2 empty lines), you get an error because both keys are the same (empty string). Why change this ?

Reply:
tso.id is undefined as tso is now the string of the tso, I think a better way is to use the same component as the country selection in gridactions